omegaconf/nodes.py

```diff
@@ -171,7 +171,11 @@ def _validate_and_convert_impl(self, value: Any) -> str:
 
         if OmegaConf.is_config(value) or is_primitive_container(value):
             raise ValidationError("Cannot convert '$VALUE_TYPE' to string: '$VALUE'")
-        return str(value)
+        if type(value) == bytes:
+            val = value.hex()
+        else:
+            val = value
+        return str(val)
```

Comment:
This change is not obvious to me.
In vanilla python we have straightforward conversion of bytes to strings:
```python
b = b"abc123"
s = "abc123"
assert str(b) == s
```
Using the `bytes.hex` method here is counterintuitive to me. What is the motivation?